bigtop-packages/src/common/impala/install_impala.sh

```diff
@@ -154,12 +154,12 @@ function symlink_lib() {
     rm $file
     base=`basename $file`
     versionless=${base/-[0-9].*/.jar}
-    ln -s ../../$dir/`basename $versionless` $DEPENDENCY_DIR/
+    ln -s $dir/`basename $versionless` $DEPENDENCY_DIR/
 }
 # Remove MR1 Hive shim
 rm -f $DEPENDENCY_DIR/hive-shims-0.23*.jar $DEPENDENCY_DIR/hadoop-core*.jar;
-for file in $DEPENDENCY_DIR/libhdfs*.so*; do symlink_lib $file ../${NATIVE_LIB_DIR}; done
-for file in $DEPENDENCY_DIR/libhadoop*.so*; do symlink_lib $file hadoop/lib/native; done
+for file in $DEPENDENCY_DIR/libhdfs*.so*; do symlink_lib $file /usr/hdp/current/hadoop-client/../usr/lib; done
```

Comment:
So, we actually can't get rid of the `..`. This is getting into symlink magic voodoo, but I'll try to explain.

The path `/usr/hdp/current/hadoop-client` is a symlink to `/usr/hdp/2.3.0.0-2557/hadoop`. The `2.3.0.0-2557` part changes based on the exact version of HDP installed. So, when I say `/usr/hdp/current/hadoop-client/../usr/lib`, that actually resolves to `/usr/hdp/2.3.0.0-2557/usr/lib` That was the only way I could find to end up with a symlink to something under `/usr/hdp/2.3.0.0-2557` without hard coding the version number.

Summary: Computers are terrible.